docs/REQFILE.md

```diff
@@ -0,0 +1,115 @@
+# Requirement Files Guide
+
+## General
+1. **Requirement files** hold the details of the team you're trying to optimize.
+2. You can optimize up to **10 cookies** in each file.
+3. Use `!reqdownload` to download a default file and view an example.
+
+## Requirements
+4. Requirements are specified under each cookie in a **yaml file**.
+5. Each cookie can accept a collection of **validity requirements** and one **objective requirement**.
+
+## Bot Moderator Permissions and Capabilities
+
+### Actions on Behalf of Others
+1. **Uploading Requirements for Users**: Moderators can upload requirement files directly into other users' inventories.
+2. **Optimizing Using User's Topping Inventory**: Moderators have the capability to optimize configurations using the topping inventory of other users.
+
+### Copying Requirement Files
+1. **Copying Files**: Moderators can copy requirement files between users, ensuring efficient management and sharing of configurations within the guild.
+
+### Code Implementation
+- Checks for `moderator_only` ensure these actions are performed securely and appropriately within the bot's framework.
+
+## Validity
+Validity requirements specify the conditions that must be met for a cookie to be considered valid:
+- **Simple**: Compare a substat to a benchmark. 
+  - Example: `Cooldown >= 6.3`
+- **Relative**: Compare a substat above or below a target.
+  - Example: `ATK SPD below Rye`
+  - Note: Targets must be other cookies listed in the file.
+- **Range**: Specify a range of valid substat values.
+  - Example: `28 <= Cooldown <= 28.5`
+- **Equality**: Pin a substat to a specific value.
+  - Example: `Cooldown == 28.3`
+
+## Objective
+Objective requirements define a goal to achieve with the cookie's stats:
+- **Objective**: Specify a goal with a substat.
+- **Special**: Include special requirements with relevant information.
+  - `E[DMG]`: Combines CRIT% and ATK.
+  - `Combo`: Combines multiple substats.
+
+## Modifiers
+Modifiers allow the inclusion of additional factors that affect the final statistics of cookies:
+- **Format**: Modifiers are defined in a separate section under `modifiers`.
+- **Types**: Various stat modifiers can be included such as CRIT%, CRIT DMG, ATK MULT, etc.
+- **Specification**: Each modifier should specify its source and value.
+
+## Discord Commands
+- **Upload a file**: `!req upload`
+  - Attach your .yaml file to the Discord message.
+  - Use `!req upload [user]` to upload a file for someone else. If no user is specified, it will automatically add the reqfile to you.
+- **View files**: `!req view`
+  - Displays all of your uploaded requirement files.
+- **Copy a file**: `!req copy [user]`
```

Comment:
Same as above, only mods need to specify a user. All of the req commands that allow an optional target assume that the target is specified by a mod